nav2_controller/src/nav2_controller.cpp

```diff
@@ -297,6 +297,12 @@ void ControllerServer::computeControl()
         return;
       }
 
+      // Don't compute a trajectory until costmap is valid (after clear costmap)
+      rclcpp::Rate r(100);
+      while (!costmap_ros_->isCurrent()) {
+        r.sleep();
```

Comment:
I'm not generally familiar with the entire threading model for Nav2 yet, but I don't like this spinning until its ready. Is there a reason for not publishing zero velocity and just trying again next time?

Reply:
Publishing zero could have jerky motion and that analog doesn't really work for the planner.

Reply:
Either way, the controller server just spins at the requested spin rate, so waiting here or skipping and updating the next update cycle (without sending a zero command) really wouldn't have much of a difference. It just helps so that if you're spinning at 20 hz and there's an update, rather than getting 19/20 out, you might still get 20/20 out but one of them had a little jitter to it. I'd think we prefer jitter over missing update - but that's open for argument.

What do you think?